source/elements/oneTBB/source/algorithms/blocked_ranges/blocked_rangeNd_cls.rst

```diff
@@ -0,0 +1,167 @@
+.. SPDX-FileCopyrightText: 2019-2024 Intel Corporation
+.. SPDX-FileCopyrightText: Contributors to the oneAPI Specification project.
+..
+.. SPDX-License-Identifier: CC-BY-4.0
+
+===============
+blocked_rangeNd
+===============
+**[algorithms.blocked_rangeNd]**
+
+Class template that represents a recursively divisible N-dimensional half-open interval.
+
+A ``blocked_rangeNd`` is the N-dimensional extension of ``blocked_range``, ``blocked_range2d`` and ``blocked_range3d``.
+We can interpret it as a Cartesian product of N instances of ``blocked_range``.
+Different to ``blocked_range2d`` and ``blocked_range3d``, all ranges have to be specified over the same type ``Value``.
+Therefore, ``blocked_rangeNd<int,2>`` is a specialisation of ``blocked_range2d`` where both dimensions have int type.
+
+.. code:: cpp
+
+    namespace oneapi {
+        namespace tbb {
+            template<typename Value, unsigned int N>
+            class blocked_rangeNd {
+            public:
+                // Types
+                using value_type = Value;
+                using dim_range_type = blocked_range<value_type>;
+                using size_type = typename dim_range_type::size_type;
+
+                // Constructors
+                blocked_rangeNd(const dim_range_type& dim0 /*, ... - exactly N parameters of the same type*/);
+                blocked_rangeNd(value_type size[N], size_type grainsize = 1);
```

Comment:
As far as I know, the parameter of an array type will not match to a braced-init-list argument, so `blocked_rangeNd<...>(/*sizes*/{100,200}, /*grainsize*/10)` will not compile. If we want it to compile, we'd need to change the parameter to e.g. `std::array<value_type,N>` or add another constructor.

Reply:
It seems I was wrong. The problem is not with the array type vs std::array, but with the by-value parameter declaration.
And the following should work: `blocked_rangeNd(const value_type (&size)[N], size_type grainsize = 1);`

See https://godbolt.org/z/Maer5Kecs for some prototyping.